lib/galaxy/managers/jobs.py

```diff
@@ -105,7 +106,9 @@ def __init__(self, app: StructuredApp):
         self.app = app
         self.dataset_manager = DatasetManager(app)
 
-    def index_query(self, trans, payload: JobIndexQueryPayload) -> sqlalchemy.engine.Result:
+    def index_query(
+        self, trans: ProvidesUserContext, payload: JobIndexQueryPayload
+    ) -> typing.Optional[sqlalchemy.engine.Result]:
```

Comment:
I'd add `from typing import Optional` and use just `Optional` here, in preparation for more type annotations.